AutoDarkModeApp/ViewModels/WallpaperPickerViewModel.cs

```diff
@@ -36,21 +84,153 @@ public WallpaperPickerViewModel(IErrorService errorService)
 
         StateUpdateHandler.OnConfigUpdate += HandleConfigUpdate;
         StateUpdateHandler.StartConfigWatcher();
+
+        PickFileCommand = new RelayCommand(async () =>
+        {
+            var openPicker = new Windows.Storage.Pickers.FileOpenPicker()
+            {
+                ViewMode = Windows.Storage.Pickers.PickerViewMode.Thumbnail,
+                SuggestedStartLocation = Windows.Storage.Pickers.PickerLocationId.PicturesLibrary,
+                FileTypeFilter = { ".jpg", ".jpeg", ".bmp", ".dib", ".png", ".jff", ".jpe", ".gif", ".tif", ".tiff", ".wdp", ".heic", ".heif", ".heics", ".heifs", ".hif", ".avci", ".avcs", ".avif", ".avifs", ".jxr", ".jxl" }
+            };
+            var hWnd = WinRT.Interop.WindowNative.GetWindowHandle(App.MainWindow);
+            WinRT.Interop.InitializeWithWindow.Initialize(openPicker, hWnd);
+            var file = await openPicker.PickSingleFileAsync();
+            if (file != null)
+            {
+                GlobalWallpaperPath = file.Path;
+            }
+        });
     }
 
     private void LoadSettings()
     {
         IsWallpaperSwitchEnabled = _builder.Config.WallpaperSwitch.Enabled;
-        if (App.Current.RequestedTheme == Microsoft.UI.Xaml.ApplicationTheme.Light)
+        SelectIndexWallpaperFillingWay = _builder.Config.WallpaperSwitch.Component.Position switch
+        {
+            WallpaperPosition.Fill => 0,
+            WallpaperPosition.Fit => 1,
+            WallpaperPosition.Stretch => 2,
+            _ => 0
+        };
+
+        ResetAllOptions();
+
+        // Generate a list with all installed Monitors, select the first one
+        List<MonitorSettings> monitors = _builder.Config.WallpaperSwitch.Component.Monitors;
+        List<MonitorSettings> disconnected = new();
+        List<MonitorSettings> connected = monitors.Where(m =>
+        {
+            // Preload tostring to avoid dropdown opening lag
+            m.ToString();
+            // Return monitors connecte to system connected monitors
+            if (m.Connected)
+            {
+                return true;
+            }
+            disconnected.Add(m);
+            return false;
+        }).ToList();
+
+        disconnected.ForEach(m =>
         {
-            SelectIndexWallpaperMode = 0;
+            m.MonitorString = $"{"DisplayMonitorDisconnected".GetLocalized()} - {m.MonitorString}";
+        });
+
+        monitors.Clear();
+        monitors.AddRange(connected);
+        monitors.AddRange(disconnected);
+        Monitors = monitors;
+
+        if (Monitors.Count > 0)
+        {
+            SelectMonitor = Monitors[0];
+        }
+
+        if (SelectIndexWallpaperMode == 0)
+        {
+            switch (_builder.Config.WallpaperSwitch.Component.TypeLight)
+            {
+                case WallpaperType.Global:
+                    IsPictureType = true;
```

Comment:
What do these variables do? Why do they exist, when they _seem_ to do the same as the previous variable?